qa/tasks/cephfs/test_scrub.py

```diff
@@ -176,3 +177,40 @@ def test_scrub_backtrace(self):
 
     def test_scrub_dup_inode(self):
         self._scrub(DupInodeWorkload(self, self.fs, self.mount_a))
+
+    def test_backtrace_health_status_post_repair(self):
+        """
+        That the MDS can repair an inodes backtrace in the data pool
+        and damage health status is updated, if it is found to be damaged.
+        """
+        # Create a file for subsequent checks
+        self.mount_a.run_shell(["mkdir", "parent_a"])
+        self.mount_a.run_shell(["touch", "parent_a/alpha"])
+        file_ino = self.mount_a.path_to_ino("parent_a/alpha")
+
+        # That backtrace and layout are written after initial flush
+        self.fs.mds_asok(["flush", "journal"])
+        backtrace = self.fs.read_backtrace(file_ino)
+        self.assertEqual(['alpha', 'parent_a'],
+                         [a['dname'] for a in backtrace['ancestors']])
+
+        # Go corrupt the backtrace
+        self.fs._write_data_xattr(file_ino, "parent",
```

Comment:
I think you should record and assert the damage count before this line

Reply:
Thanks, I did by mistake.